### ISSUE TITLE
Replace Redux with Context API and Hooks

### ISSUE THREAD

**Issue by danielgrijalva** [danielgrijalva/overworld#160]
This proposal would simplify the code and the overall front end code. We would need to remove all the Redux code (actions, reducers, store…) and replace it with [Context](https://reactjs.org/docs/context.html), which is simpler and easier to understand. 

[Here’s](https://medium.com/octopus-labs-london/replacing-redux-with-react-hooks-and-context-part-1-11b72ffdb533) an article that explains how to achieve this.

**Comment by detri** [comment]
I think this is potentially blocked by #156. The front end is still using class components and we need to be using FCs to be able to use hooks. Of course, I suppose it is possible to refactor only the components using the store to be FCs.

Also, I don't think all of the redux code needs to be removed. The actions and reducers can remain, we just need to turn the store into a context.

Just my two cents though 😄

**Comment by danielgrijalva** [comment]
@detri you’re right about the actions and reducers; we can leave them and utilize them with Context 😅

And yeah I suppose we need to refactor the class components into functional ones first… I’m thinking that this issue is too big (refactor into Hooks _and_ implement Context) so I will probably split it into two.

Let me know if you are interested in working on this!

**Comment by detri** [comment]
Sure - I'm looking to get a start into contributing to open source and this seems like a good project for it. I wouldn't mind tackling turning components into functional ones. That is by far the more time-consuming task -- would it be advantageous to split that up into, say, one PR per module? It would ensure everything works together and make things easier on reviewers (500 line PRs are no fun to review)

**Comment by danielgrijalva** [comment]
One PR per module/component sounds great!

I’ve put up #172 for this matter so, refer to that issue for your future PRs. Thanks @detri!

**Comment by ryaweb** [comment]
Hi are you still looking for help on this

**Comment by joshnho** [comment]
still open?

**Comment by ThomasMaher** [comment]
Would also like to contribute to this if its still open.
